backend/sirius-web-spring-collaborative/src/main/java/org/eclipse/sirius/web/spring/collaborative/projects/EditingContextEventProcessorParameters.java

```diff
@@ -0,0 +1,171 @@
+
+        private IEditingContext editingContext;
+
+        private IEditingContextPersistenceService editingContextPersistenceService;
+
+        private ApplicationEventPublisher applicationEventPublisher;
+
+        private List<IEditingContextEventHandler> editingContextEventHandlers;
+
+        private IRepresentationEventProcessorComposedFactory representationEventProcessorComposedFactory;
+
+        private IDanglingRepresentationDeletionService danglingRepresentationDeletionService;
+
+        private IEditingContextEventProcessorExecutorServiceProvider executorServiceProvider;
+
+        private Builder() {
+            // Prevent instantiation
+        }
+
+        public Builder messageService(ICollaborativeMessageService messageService) {
+            this.messageService = Objects.requireNonNull(messageService);
+            return this;
+        }
+
+        public Builder editingContext(IEditingContext editingContext) {
+            this.editingContext = Objects.requireNonNull(editingContext);
+            return this;
+        }
+
+        public Builder editingContextPersistenceService(IEditingContextPersistenceService editingContextPersistenceService) {
+            this.editingContextPersistenceService = Objects.requireNonNull(editingContextPersistenceService);
+            return this;
+        }
+
+        public Builder applicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+            this.applicationEventPublisher = Objects.requireNonNull(applicationEventPublisher);
+            return this;
+        }
+
+        public Builder editingContextEventHandlers(List<IEditingContextEventHandler> editingContextEventHandlers) {
+            this.editingContextEventHandlers = Objects.requireNonNull(editingContextEventHandlers);
+            return this;
+        }
+
+        public Builder representationEventProcessorComposedFactory(IRepresentationEventProcessorComposedFactory representationEventProcessorComposedFactory) {
+            this.representationEventProcessorComposedFactory = Objects.requireNonNull(representationEventProcessorComposedFactory);
+            return this;
+        }
+
+        public Builder danglingRepresentationDeletionService(IDanglingRepresentationDeletionService danglingRepresentationDeletionService) {
+            this.danglingRepresentationDeletionService = Objects.requireNonNull(danglingRepresentationDeletionService);
+            return this;
+        }
+
+        public Builder executorServiceProvider(IEditingContextEventProcessorExecutorServiceProvider executorServiceProvider) {
+            this.executorServiceProvider = Objects.requireNonNull(executorServiceProvider);
+            return this;
+        }
+
+        public EditingContextEventProcessorParameters build() {
+            EditingContextEventProcessorParameters parameters = new EditingContextEventProcessorParameters();
+            parameters.messageService = Objects.requireNonNull(this.messageService);
```

Comment:
Why there needs the not-null assertion when it is never null?

Reply:
Someone could forget to set it, it would thus be null